packages/backend/src/open_payments/payment/incoming/model.ts

```diff
@@ -194,47 +198,24 @@ export class IncomingPayment
     return data
   }
 
-  public $beforeInsert(context: QueryContext): void {
-    super.$beforeInsert(context)
-    this.connectionId = this.connectionId || uuid()
-  }
-
-  public $beforeUpdate(opts: ModelOptions, queryContext: QueryContext): void {
-    super.$beforeUpdate(opts, queryContext)
-    if (
-      [IncomingPaymentState.Completed, IncomingPaymentState.Expired].includes(
-        this.state
-      )
-    ) {
-      this.connectionId = null
-    }
-  }
-
   public toOpenPaymentsType(
     walletAddress: WalletAddress
   ): OpenPaymentsIncomingPayment
   public toOpenPaymentsType(
     walletAddress: WalletAddress,
-    ilpStreamConnection: Connection
-  ): OpenPaymentsIncomingPaymentWithConnection
+    ilpStreamCredentials: IlpStreamCredentials
+  ): OpenPaymentsIncomingPaymentWithPaymentMethod
   public toOpenPaymentsType(
     walletAddress: WalletAddress,
-    ilpStreamConnection: string
-  ): OpenPaymentsIncomingPaymentWithConnectionUrl
-  public toOpenPaymentsType(
-    walletAddress: WalletAddress,
-    ilpStreamConnection?: Connection | string
-  ):
-    | OpenPaymentsIncomingPaymentWithConnection
-    | OpenPaymentsIncomingPaymentWithConnectionUrl
+    ilpStreamCredentials?: IlpStreamCredentials
+  ): OpenPaymentsIncomingPayment | OpenPaymentsIncomingPaymentWithPaymentMethod
 
   public toOpenPaymentsType(
     walletAddress: WalletAddress,
-    ilpStreamConnection?: Connection | string
+    ilpStreamCredentials?: IlpStreamCredentials
```

Comment:
To spitball a future idea for this, maybe this parameter could be genericized into `method` which has a type that all payment methods implement and can call a `.toOpenPayments()` function.

Reply:
Im not sure if I'm following but wanted to note there have been some recent changes that might effect this suggestion: https://github.com/interledger/rafiki/pull/2054#discussion_r1367369897